src/wazuh_db/wdb.h

```diff
@@ -1945,13 +1945,29 @@ int wdb_global_sync_agent_info_set(wdb_t *wdb, cJSON *agent_info);
  *              WDB_GROUP_SYNC_STATUS for agents tagged as sync_req,
  *              WDB_GROUP_CKS_MISMATCH for agents with difference between the CKS in the master and the worker.
  * @param [in] last_agent_id ID where to start querying.
+ * @param [in] set_synced Indicates if the obtained groups must be set as synced.
+ * @param [in] get_hash Indicates if the response must append the group_hash once all the groups have been obtained.
  * @param [out] output A buffer where the response is written. Must be de-allocated by the caller.
  * @return wdbc_result to represent if all agents has being obtained.
  */
 wdbc_result wdb_global_sync_agent_groups_get(wdb_t *wdb,
                                              wdb_groups_sync_condition_t condition,
                                              int last_agent_id,
+                                             bool set_synced,
+                                             bool get_hash,
                                              char **output);
+/**
+ * @brief Function to update group_sync_status of a particular agent.
+ *
+ * @param [in] wdb The Global struct database.
+ * @param [in] id The agent ID
+ * @param [in] sync_status The value of sync_status
+ * @return OS_SUCCESS On success. OS_ERROR On error.
+ */
+int wdb_global_set_groups_sync_status(wdb_t *wdb,
```

Comment:
**wdb_global_sync_agent_groups_get()** is preceded by "sync" word because it´s mainly a job of syncing (the get part of the sync), on the other hand **wdb_global_set_groups_sync_status** is setting the **sync_status** column.
But, it´s true that other similar methods are named with **agent_groups** as **wdb_global_set_agent_groups** and **wdb_global_set_agent_group_context**. So, this method should be named **wdb_global_set_agent_groups_sync_status**